raster-foundry/raster-foundry
# Enable team creation on user's team list page

## Overview

This PR:
 * enables creating new teams on an user's team list page (page route `/user/me/teams`)
 * adds an optional boolean query parameter `withGroupName` to the `users/me/roles` endpoint indicating whether to make user group roles come back with platform, organization, and team names.

### Checklist


### Demo

<img width="1226" alt="screen shot 2018-12-03 at 4 50 18 pm" src="https://user-images.githubusercontent.com/16109558/49405312-0d5ccd80-f720-11e8-9e41-a304bcc203e7.png">

Note that if you are on an organization's team list page (an UI route like `admin/organization/<orgId>/teams`), the first dropdown for choosing an organization to add the team to should not show up.


## Testing Instructions

 * Log in as a member of a platform (as opposed to an admin of it, since admins of either a platform or an organization won't be added to the team to be created afterwards automatically per [these lines](https://github.com/raster-foundry/raster-foundry/blob/develop/app-backend/api/src/main/scala/platform/Routes.scala#L531-L540))
 * Go to "My Teams" page, which goes to the UI for route `/user/me/teams`
 * Check if there is a "Create team" button
 * Clicking on this button should bring up a team create modal with a dropdown to select an organization
 * Choose an organization (if you'd like to be added to this team being created, choose an organization you are not an admin of, per reasons mentioned in step 1)
 * Provide a name and click on create team button
 * Page state should update so that you will see teams that you are a member or an admin of

Closes https://github.com/raster-foundry/raster-foundry/issues/4311
